shared/stories/prop-providers.js

```diff
@@ -20,60 +12,66 @@ const compose = (...providers: any[]) => {
 
-const Usernames = (following: string[], you?: string) => ({
-  Usernames: (props: any) => {
-    const {usernames} = props
+const defaultYou = 'ayoubd'
+const defaultFollowing = ['max', 'cnojima', 'cdixon']
+const defaultFollowers = ['max', 'akalin']
+
+const Usernames = (following: string[] = defaultFollowing, you: string = defaultYou) => ({
+  Usernames: (ownProps: _Usernames.ConnectedProps): _Usernames.Props => {
+    const {usernames} = ownProps
     const users = (usernames || []).map(username => ({
       username,
       following: following.includes(username),
       you: you ? username === you : false,
     }))
     return {
-      ...props,
+      ...ownProps,
       users,
       onUsernameClicked: action('onUsernameClicked'),
     }
   },
 })
 
 const WaitingButton = () => ({
-  WaitingButton: (props: any) => props,
+  WaitingButton: (ownProps: _WaitingButton.OwnProps): _WaitingButton.Props => ({
+    ...ownProps,
+    storeWaiting: false,
+  }),
 })
 
-const Avatar = (following: string[], followers: string[]) => ({
-  Avatar: (props: any) =>
-    mockOwnToViewProps(props, following.includes(props.username), followers.includes(props.username)),
+const Avatar = (following: string[] = defaultFollowing, followers: string[] = defaultFollowers) => ({
+  Avatar: (ownProps: _Avatar.OwnProps) => _Avatar.mockOwnToViewProps(ownProps, following, followers, action),
 })
 
 const TeamDropdownMenu = (adminTeams?: string[], teamMemberCounts?: {[key: string]: number}) => ({
-  TeamDropdownMenu: (props: any) => ({
-    _hasCanPerform: true,
-    _loadOperations: unexpected('_loadOperations'),
-    attachTo: props.attachTo,
+  TeamDropdownMenu: (ownProps: _TeamDropdownMenu.OwnProps): _TeamDropdownMenu.Props => ({
+    _loadOperations: action('_loadOperations'),
+    attachTo: ownProps.attachTo,
     badgeSubscribe: false,
-    canAddPeople: (adminTeams && adminTeams.includes(props.teamname)) || true,
-    isSmallTeam: props.isSmallTeam,
-    memberCount: (teamMemberCounts && teamMemberCounts[props.teamname]) || 100,
-    teamname: props.teamname,
-    visible: props.visible,
+    canAddPeople: (adminTeams && adminTeams.includes(ownProps.teamname)) || true,
+    isSmallTeam: ownProps.isSmallTeam,
+    manageChannelsSubtitle: ownProps.isSmallTeam ? 'Turns this into a big team' : '',
+    manageChannelsTitle: ownProps.isSmallTeam ? 'Create chat channels...' : 'Manage chat channels',
+    memberCount: (teamMemberCounts && teamMemberCounts[ownProps.teamname]) || 100,
+    teamname: ownProps.teamname,
+    visible: ownProps.visible,
     onAddPeople: action('onAddPeople'),
-    onHidden: props.onHidden,
+    onHidden: ownProps.onHidden,
     onInvite: action('onInvite'),
     onLeaveTeam: action('onLeaveTeam'),
     onManageChannels: action('onManageChannels'),
     onViewTeam: action('onViewTeam'),
   }),
 })
 
-const Common = () => compose(
-  Usernames(['max', 'cnojima', 'cdixon'], 'ayoubd'),
-  Avatar(['following', 'both'], ['followers', 'both']),
```

Comment:
these are supposed to be usernames, so may as well make them consistent with the argument to Usernames